DebugInfomationFormat.md

```diff
@@ -0,0 +1,220 @@
+#Debug Symbol Integration
+
+This is a design proposal for providing debug information within WebAssembly
+that would allow source-level debugging of code compiled into WebAssembly.
+
+## Goals and Principles
+
+The immediate goal is to implement enough debugging functionality in WebAssembly
+MVP to let users perform at least rudimentary debugging on their source.
+Another goal is not to bake in any limitations that would prevent reasonable
+future evolution of the standard.
+
+Specifically, we are currently not focused on identifying a language-independent
+subset of the debug format -- this proposal bundles language-dependent and
+language-independent parts together.
+
+META: We need an evolution strategy that allows new front-end/debugger pairs to
+use the format in the future to transfer information currently unanticipated.
+Examples: a) dynamic scoping in Lisp; b) full DWARF 4 equivalence.
+
+## Debug Info Goes into Extra Sections
+
+To allow easy stripping of debug info, all of it will go into a separate
+section.  In the binary format, this will be an unknown section inserted into
+the wasm file.  Because this new info will make the current
+[name section](BinaryEncoding.md#name-section) redundant, we propose deprecating
+the name section.
+
+META: How do we distinguish the debug-containing unknown section from any other
+unknown section?
+
+In the text format, this will be an extra child of the **module** element named
+**debug**.
+
+META: If there's a better name than **debug**, we can certainly use it.  Also,
+currently undecided about allowing multiple **debug** children, like **func**.
```

Comment:
At an initial read, this sounds like a plan to make a format that has the eventual goal of reimplementing the functionality of DWARF. It's not explicitly spelled out here, but I assume the purpose of standardizing such a format is to allow browsers to consume it and provide their own HLL debugging experiences. Is this the vision here?

If so, I'd like to have a discussion about the merits of that vision compared with the merits of another one:

Another vision is that WebAssembly just provides '@' notation and byte offsets for identifying locations in WebAssembly code (as you describe), browser APIs for debugging primitives like reading out memory, setting breakpoints, and so on (in a security-appropriate manner), and browser APIs for reading the contents of unknown sections from wasm files. These make it possible for content to implement debugging functionality itself, sch as in the form of a library that gets linked in, or possibly in the form of code that the user might request be loaded along side the main content of a site.